sklearn/utils/estimator_checks.py

```diff
@@ -1443,41 +1443,90 @@ def check_supervised_y_2d(name, estimator_orig):
     assert_allclose(y_pred.ravel(), y_pred_2d.ravel())
 
 
-@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+@ignore_warnings
+def check_classifiers_predictions(X, y, name, classifier_orig):
+    classes = np.unique(y)
+    classifier = clone(classifier_orig)
+    if name == 'BernoulliNB':
+        X = X > X.mean()
+    set_random_state(classifier)
+
+    classifier.fit(X, y)
+    y_pred = classifier.predict(X)
+
+    if hasattr(classifier, "decision_function"):
+        decision = classifier.decision_function(X)
+        n_samples, n_features = X.shape
+        assert(isinstance(decision, np.ndarray))
+        if len(classes) == 2:
+            dec_pred = (decision.ravel() > 0).astype(np.int)
+            dec_exp = classifier.classes_[dec_pred]
+            assert_array_equal(dec_exp, y_pred,
+                               err_msg="decision_function does not match "
+                               "classifier for %r: expected '%s', got '%s'" %
+                               (classifier, ", ".join(map(str, dec_exp)),
+                                ", ".join(map(str, y_pred))))
+        elif getattr(classifier, 'decision_function_shape', 'ovr') == 'ovr':
+            decision_y = np.argmax(decision, axis=1).astype(int)
+            y_exp = classifier.classes_[decision_y]
+            assert_array_equal(y_exp, y_pred,
+                               err_msg="decision_function does not match "
+                               "classifier for %r: expected '%s', got '%s'" %
+                               (classifier, ", ".join(map(str, y_exp)),
+                                ", ".join(map(str, y_pred))))
+
+    # training set performance
+    if name != "ComplementNB":
+        # This is a pathological data set for ComplementNB.
+        assert_array_equal(np.unique(y), np.unique(y_pred))
+    if np.any(classifier.classes_ != classes):
+        assert_array_equal(classes, classifier.classes_,
+                           err_msg="Unexpected classes_ attribute for %r: "
+                           "expected '%s', got '%s'" %
+                           (classifier, ", ".join(map(str, classes)),
+                            ", ".join(map(str, classifier.classes_))))
+
+
+def choose_check_classifiers_labels(name, y, y_names):
+    if name in ["LabelPropagation", "LabelSpreading"]:
+        # TODO some complication with -1 label
+        return y
+
+    return y_names
+
+
 def check_classifiers_classes(name, classifier_orig):
```

Comment:
It seems that we could use parametrize using pytest.
Do we avoid to do that in the estimator_checks tests to not force pytest as a dependency, @jnothman @lesteve?